app/api/discount_codes.py

```diff
@@ -271,8 +272,12 @@ def before_get(self, args, kwargs):
             )
             if discount:
                 kwargs['id'] = discount.id
-                discount_tz = discount.valid_from.tzinfo
-                current_time = datetime.now().replace(tzinfo=discount_tz)
+                if discount.valid_from:
+                    discount_tz = discount.valid_from.tzinfo
+                if discount_tz:
+                    current_time = datetime.now().replace(tzinfo=discount_tz)
+                else:
+                    current_time = datetime.now().replace(tzinfo=timezone('UTC'))
```

Comment:
```suggestion
                current_time = datetime.now().replace(tzinfo=discount_tz or timezone('UTC'))
```

Reply:
@iamareebjamal Updated mr :v: Thanks for the suggestion!